### PR TITLE
wafv2_ip_set - add support for updating tags

### PR DESCRIPTION
**Pull request by tremble**
##### SUMMARY

- Added support for purge_tags
- Added tags to return values
- Added support for updating tags
- Moved to common docs_fragment for tagging

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

plugins/module_utils/wafv2.py
plugins/modules/wafv2_ip_set.py
plugins/modules/wafv2_ip_set_info.py

##### ADDITIONAL INFORMATION

Depends on : https://github.com/mattclay/aws-terminator/pull/213

### CODE REVIEW

**Comment by gravesm** [comment]
recheck

**Review by alinabuzachis — APPROVED** [review]

**Review by markuman — APPROVED** [review]

**Comment by markuman** [comment]
there should be a `note: ` in the docs section, that `purge_tags` is supported since 4.0.0.  
And it must be set to `defaults(no)` in the first release

File: plugins/modules/wafv2_ip_set.py [hunk 1]
```diff
@@ -275,8 +281,9 @@ def main():
         description=dict(type='str'),
         ip_address_version=dict(type='str', choices=['IPV4', 'IPV6']),
         addresses=dict(type='list', elements='str'),
-        tags=dict(type='dict'),
-        purge_addresses=dict(type='bool', default=True)
+        tags=dict(type='dict', aliases=['resource_tags']),
+        purge_tags=dict(type='bool', default=True),
```
Comment (markuman) [inline plugins/modules/wafv2_ip_set.py:285]:
```suggestion
        purge_tags=dict(type='bool', default=False),
```

in the first release

**Comment by tremble** [comment]
> there should be a `note: ` in the docs section, that `purge_tags` is supported since 4.0.0. And it must be set to `defaults(no)` in the first release

I was in two minds over the default value.  Since it didn't seem to set tags at all on update previously, I went with defaulting to True...

**Comment by markuman** [comment]
> > there should be a `note: ` in the docs section, that `purge_tags` is supported since 4.0.0. And it must be set to `defaults(no)` in the first release
> 
> I was in two minds over the default value. Since it didn't seem to set tags at all on update previously, I went with defaulting to True...

Yes good question. This will hit only users that are dealing with boto3 and maybe other orchestration tools like cloudformation....but dunno. better safe than sorry?

**Comment by tremble** [comment]
> Yes good question. This will hit only users that are dealing with boto3 and maybe other orchestration tools like cloudformation....but dunno. better safe than sorry?

unless tags is explicitly set in the task `ensure_wafv2_tags` will do nothing, so we should be good...

**Pull request merged**